File: mobile/src/main/java/treehou/se/habit/ui/sitemaps/page/PageContract.java
package treehou.se.habit.ui.sitemaps.page;


import java.util.List;

import se.treehou.ng.ohcommunicator.connector.models.OHLinkedPage;
import treehou.se.habit.mvp.BasePresenter;
import treehou.se.habit.mvp.BaseView;
import treehou.se.habit.ui.widgets.WidgetFactory;

public interface PageContract {
    String ARG_PAGE    = "ARG_PAGE";
    String ARG_SERVER  = "ARG_SERVER";
    String STATE_PAGE = "STATE_PAGE";

    interface View extends BaseView<Presenter> {
        void showLostServerConnectionMessage();
        void closeView();
        void updatePage(OHLinkedPage page);
        void setWidgets(List<WidgetFactory.IWidgetHolder> widgets);
    }

    interface Presenter extends BasePresenter {

    }
}

File: mobile/src/main/java/treehou/se/habit/ui/sitemaps/page/PageContract.kt
package treehou.se.habit.ui.sitemaps.page


import se.treehou.ng.ohcommunicator.connector.models.OHLinkedPage
import se.treehou.ng.ohcommunicator.connector.models.OHWidget
import treehou.se.habit.mvp.BasePresenter
import treehou.se.habit.mvp.BaseView

interface PageContract {

    interface View : BaseView<Presenter> {
        fun showLostServerConnectionMessage()
        fun closeView()
        fun updatePage(page: OHLinkedPage)
        fun setWidgets(widgets: List<OHWidget>)
    }

    interface Presenter : BasePresenter

    companion object {
        val ARG_PAGE = "ARG_PAGE"
        val ARG_SERVER = "ARG_SERVER"
        val STATE_PAGE = "STATE_PAGE"
    }
}
